### ISSUE TITLE
devcontainer - Error: Instance Exited before being ready and without throwing an error!

### ISSUE THREAD

**Issue by Yaikava** [nodkz/mongodb-memory-server#764]
## Versions

I am using this devcontainer `mcr.microsoft.com/devcontainers/typescript-node:20-buster`.
For some reason this is not working

- NodeJS: 20
- mongodb-memory-server-*: no idea
- mongodb(the binary version): no idea
- mongodb(the js package): 7.6.3
- system: Tried with host os of windows 10 and fedora 38, same result in docker

package: mongo-memory-server-core

## What is your question?

I am using a package which is using this so I cannot really get the exact run options. However in docker containers it is doing this. Any idea how to solve it?

Logs
```
Starting embedded database for datastores [primary]...
  MongoMS:MongoMemoryServer Mongo[unknown]: start: Called .start() method +0ms
  MongoMS:MongoMemoryServer Mongo[unknown]: _startUpInstance: Called MongoMemoryServer._startUpInstance() method +2ms
  MongoMS:MongoMemoryServer Mongo[unknown]: getStartOptions +2ms
  MongoMS:MongoMemoryServer Mongo[unknown]: getStartOptions: Checking if "/workspaces/school/project/projectname-server/target/db_data}" (no new tmpDir) already has data +7ms
  MongoMS:MongoMemoryServer Mongo[unknown]: _startUpInstance: Creating new MongoDB instance with options: {
  instance: {
    port: 27017,
    dbName: 'admin',
    ip: '127.0.0.1',
    storageEngine: 'wiredTiger',
    replSet: undefined,
    dbPath: '/workspaces/school/project/projectname-server/target/db_data',
    tmpDir: undefined,
    args: undefined,
    auth: undefined
  },
  binary: {
    version: '4.0.28',
    downloadDir: '/home/node/.cache/mongodb-binaries'
  },
  spawn: undefined
} +1ms
  MongoMS:MongoInstance create: Called .create() method +0ms
  MongoMS:MongoInstance Mongo[27017]: start +1ms
  MongoMS:MongoBinary getPath +0ms
  MongoMS:DryMongoBinary generateOptions +0ms
  MongoMS:utils tryReleaseFile: "/etc/upstream-release/lsb-release" does not exist +0ms
  MongoMS:getos getLinuxInformation: Using etcOsRelease +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +9ms
  MongoMS:DryMongoBinary generatePaths +1ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +1ms
  MongoMS:DryMongoBinary combineBinaryName +2ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +2ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28',
  modulesCache: '/workspaces/school/project/projectname-server/node_modules/.cache/mongodb-memory-server/mongod-x64-debian-4.0.28',
  relative: '/workspaces/school/project/projectname-server/mongodb-binaries/mongod-x64-debian-4.0.28',
  resolveConfig: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28'
}  +0ms
  MongoMS:DryMongoBinary generateDownloadPath: no existing binary for version "4.0.28" was found +2ms
  MongoMS:DryMongoBinary generateDownloadPath: using resolveConfig (DOWNLOAD_DIR) "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:MongoBinary getPath: MongoBinary options: {
  "version": "4.0.28",
  "downloadDir": "/home/node/.cache/mongodb-binaries",
  "os": {
    "os": "linux",
    "dist": "debian",
    "codename": "buster",
    "release": "10"
  },
  "arch": "x64",
  "systemBinary": "",
  "platform": "linux",
  "checkMD5": false
} +20ms
  MongoMS:DryMongoBinary locateBinary: Trying to locate Binary for version "4.0.28" +1ms
  MongoMS:DryMongoBinary generateOptions +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +0ms
  MongoMS:DryMongoBinary generatePaths +0ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +1ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28',
  modulesCache: '/workspaces/school/project/projectname-server/node_modules/.cache/mongodb-memory-server/mongod-x64-debian-4.0.28',
  relative: '/workspaces/school/project/projectname-server/mongodb-binaries/mongod-x64-debian-4.0.28',
  resolveConfig: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28'
}  +0ms
  MongoMS:DryMongoBinary generateDownloadPath: no existing binary for version "4.0.28" was found +1ms
  MongoMS:DryMongoBinary generateDownloadPath: using resolveConfig (DOWNLOAD_DIR) "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:DryMongoBinary locateBinary: running generateDownloadPath +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +0ms
  MongoMS:DryMongoBinary generatePaths +0ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28',
  modulesCache: '/workspaces/school/project/projectname-server/node_modules/.cache/mongodb-memory-server/mongod-x64-debian-4.0.28',
  relative: '/workspaces/school/project/projectname-server/mongodb-binaries/mongod-x64-debian-4.0.28',
  resolveConfig: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28'
}  +0ms
  MongoMS:DryMongoBinary generateDownloadPath: no existing binary for version "4.0.28" was found +5ms
  MongoMS:DryMongoBinary generateDownloadPath: using resolveConfig (DOWNLOAD_DIR) "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:DryMongoBinary locateBinary: could not find a existing binary +0ms
  MongoMS:MongoBinary getPath: "RUNTIME_DOWNLOAD" is "true", trying to download +7ms
  MongoMS:MongoBinary download +1ms
  MongoMS:MongoBinary download: Waiting to acquire Download lock for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +2ms
  MongoMS:LockFile lock: Locking file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +0ms
  MongoMS:LockFile checkLock: for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" with uuid: "undefined" +1ms
  MongoMS:LockFile createLock: trying to create a lock file for "/home/node/.cache/mongodb-binaries/4.0.28.lock" +0ms
  MongoMS:LockFile createLock: Lock File Created for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +3ms
  MongoMS:MongoBinary download: Download lock acquired +4ms
  MongoMS:MongoBinary download: Adding version 4.0.28 to cache +0ms
  MongoMS:MongoBinaryDownload getMongodPath +0ms
  MongoMS:DryMongoBinary generateOptions +8ms
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +0ms
  MongoMS:DryMongoBinary generatePaths +0ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28',
  modulesCache: '/workspaces/school/project/projectname-server/node_modules/.cache/mongodb-memory-server/mongod-x64-debian-4.0.28',
  relative: '/workspaces/school/project/projectname-server/mongodb-binaries/mongod-x64-debian-4.0.28',
  resolveConfig: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28'
}  +0ms
  MongoMS:DryMongoBinary generateDownloadPath: no existing binary for version "4.0.28" was found +1ms
  MongoMS:DryMongoBinary generateDownloadPath: using resolveConfig (DOWNLOAD_DIR) "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:MongoBinaryDownload startDownload +2ms
  MongoMS:MongoBinaryDownloadUrl getDebianVersionString: requested version "4.0.28" not available for osrelease "10", using "92" +0ms
  MongoMS:MongoBinaryDownloadUrl Using "mongodb-linux-x86_64-debian92-4.0.28.tgz" as the Archive String +0ms
  MongoMS:MongoBinaryDownloadUrl Using "https://fastdl.mongodb.org" as the mirror +0ms
  MongoMS:MongoBinaryDownload download +3ms
  MongoMS:MongoBinaryDownload download: Downloading: "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-debian92-4.0.28.tgz" +0ms
  MongoMS:MongoBinaryDownload httpDownload +0ms
  MongoMS:MongoBinaryDownload httpDownload: trying to download "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-debian92-4.0.28.tgz" +0ms
  MongoMS:MongoBinaryDownload httpDownload: moved "/home/node/.cache/mongodb-binaries/mongodb-linux-x86_64-debian92-4.0.28.tgz.downloading" to "/home/node/.cache/mongodb-binaries/mongodb-linux-x86_64-debian92-4.0.28.tgz" +31s
  MongoMS:MongoBinaryDownload makeMD5check: Checking MD5 of downloaded binary... +0ms
  MongoMS:MongoBinaryDownload makeMD5check: checkMD5 is disabled +1ms
  MongoMS:MongoBinaryDownload extract +0ms
  MongoMS:DryMongoBinary generateOptions +31s
  MongoMS:DryMongoBinary generateDownloadPath: Generating Download Path, preferGlobal: "true" +0ms
  MongoMS:DryMongoBinary generatePaths +1ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +1ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generatePaths: resolveConfigValue is not empty +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:DryMongoBinary generateDownloadPath: Paths: {
  legacyHomeCache: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28',
  modulesCache: '/workspaces/school/project/projectname-server/node_modules/.cache/mongodb-memory-server/mongod-x64-debian-4.0.28',
  relative: '/workspaces/school/project/projectname-server/mongodb-binaries/mongod-x64-debian-4.0.28',
  resolveConfig: '/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28'
}  +0ms
  MongoMS:DryMongoBinary generateDownloadPath: no existing binary for version "4.0.28" was found +3ms
  MongoMS:DryMongoBinary generateDownloadPath: using resolveConfig (DOWNLOAD_DIR) "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:DryMongoBinary getBinaryName +0ms
  MongoMS:DryMongoBinary combineBinaryName +0ms
  MongoMS:MongoBinaryDownload extract: archive: "/home/node/.cache/mongodb-binaries/mongodb-linux-x86_64-debian92-4.0.28.tgz" final: "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +5ms
  MongoMS:MongoBinaryDownload extractTarGz +1ms
  MongoMS:MongoBinary download: Removing Download lock +33s
  MongoMS:LockFile unlock: Unlocking file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +33s
  MongoMS:LockFile checkLock: for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" with uuid: "42ad3168-b952-48b6-a468-77e3e66d3fe5" +0ms
  MongoMS:LockFile checkLock: Lock File Already exists, and is for *this* process, with uuid: "42ad3168-b952-48b6-a468-77e3e66d3fe5" +1ms
  MongoMS:LockFile unlock: Lock Status was "availableInstance" for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +0ms
  MongoMS:LockFile unlockCleanup: for file "/home/node/.cache/mongodb-binaries/4.0.28.lock" +1ms
  MongoMS:MongoBinary download: Download lock removed +2ms
  MongoMS:MongoBinary getPath: Mongod binary path: "/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28" +0ms
  MongoMS:MongoInstance Mongo[27017]: start: Starting Processes +33s
  MongoMS:MongoInstance Mongo[27017]: _launchMongod: Launching Mongod Process +0ms
  MongoMS:MongoInstance Mongo[27017]: prepareCommandArgs +1ms
  MongoMS:MongoInstance Mongo[27017]: prepareCommandArgs: final argument array:["--port","27017","--dbpath","/workspaces/school/project/projectname-server/target/db_data","--storageEngine","wiredTiger","--bind_ip","127.0.0.1","--noauth"] +0ms
  MongoMS:MongoInstance Mongo[27017]: _launchKiller: Launching Killer Process (parent: 809, child: 1204) +9ms
  MongoMS:MongoInstance Mongo[27017]: stderrHandler: ""/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28)
  MongoMS:MongoInstance "" +27ms
  MongoMS:MongoInstance Mongo[27017]: closeHandler: Mongod instance closed with an non-0 (or non 12 on windows) code! +2ms
  MongoMS:MongoInstance Mongo[27017]: closeHandler: "1" "null" +0ms
Error: Instance Exited before being ready and without throwing an error!
    at MongoInstance.<anonymous> (/workspaces/school/project/projectname-server/node_modules/mongodb-memory-server-core/lib/util/MongoInstance.js:140:25)
    at Object.onceWrapper (node:events:626:26)
    at MongoInstance.emit (node:events:511:28)
    at MongoInstance.closeHandler (/workspaces/school/project/projectname-server/node_modules/mongodb-memory-server-core/lib/util/MongoInstance.js:275:14)
    at ChildProcess.emit (node:events:511:28)
    at maybeClose (node:internal/child_process:1098:16)
    at Socket.<anonymous> (node:internal/child_process:456:11)
    at Socket.emit (node:events:511:28)
    at Pipe.<anonymous> (node:net:334:12)
```

**Comment by hasezoey** [comment]
your error is:
```txt
/home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/node/.cache/mongodb-binaries/mongod-x64-debian-4.0.28)
```

which basically makes this a duplicate of #204

TL;DR: your options to fix this are:
- use a newer binary version than 4.0.28 (minimal version being 4.2.1 for debian10)
- use a older version of debian (debian9)
- install openssl3 somehow (likely not in the official repositories for debian10)
